src/main/java/frc/robot/commands/PrepareAmpShot.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import frc.robot.subsystems.Elevator;
+import frc.robot.subsystems.Shooter;
+import frc.robot.subsystems.ShooterPivot;
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
+// information, see:
+// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html
+public class PrepareAmpShot extends ParallelCommandGroup {
+  /** Creates a new PrepareAmpShot. */
+  public PrepareAmpShot(Elevator elevator, ShooterPivot shooterPivot, Shooter shooter) {
+    // Add your commands in the addCommands() call, e.g.
+    // addCommands(new FooCommand(), new BarCommand());
+    addCommands(
+      new SetElevatorLength(elevator, Elevator.AMP_SCORE_LENGTH),
+      new SetPivotAngle(shooterPivot, ShooterPivot.AMP_SCORE_ANGLE_RADIANS),
+      new InstantCommand(shooter::setSpeakerShootModeFalse)
```

Comment:
Why add a custom function?  You can just call the existing set function with a parameter of "false". See PrepareSpeakerShot.

Reply:
when i try to just use the function setSpeakerShootMode and try to pass in  pram it gives an error

Reply:
Look at the syntax in PrepareSpeakerShot

Reply:
nvm found some docs for it